bigquery_etl/query_scheduling/task.py

```diff
@@ -125,6 +122,66 @@ def get_execution_delta(self, schedule_interval):
         return None
 
 
+@attr.s(auto_attribs=True, frozen=True)
+class TableSensorTask:
+    """Representation of a sensor task to wait for a table to exist."""
+
+    task_id: str = attr.ib()
```

Comment:
For other task types we are generating the `task_id` based on dataset identifiers (like `wait_for`). Should we do the same here and keep this option to overwrite the default?

Reply:
That was my initial thought as well, but ran into some complicating factors:
- There will almost certainly be Jinja in the underlying identifiers.  Some ways I considered handling that:
  - Keep a sanitized version of the Jinja in the task ID (but then we'd have awkward task ID suffixes like `_macros_ds_add_ds_1_ds_nodash`, and the task ID could get altered whenever the Jinja is tweaked).
  - Remove the Jinja and append a hash of the overall underlying value to guarantee uniqueness (but then when viewing the tasks in Airflow you couldn't easily reason about which one was which if there were multiple for the same table, and again the task ID could get altered whenever the Jinja is tweaked).
  - Remove the Jinja and append a numerical suffix like `_1`, `_2`, etc. in case there are multiple for the same table (but then the meaning of those numbered tasks could get mixed up depending on how we're ordering them and when tasks get added/removed).
- If we're auto-generating the task IDs we take on the responsibility of handling task ID conflicts (e.g. if multiple ETLs depend on the same tables/partitions). Some ways I considered handling that:
  - Only output one sensor task in cases where multiple ETLs depend on the same tables/partitions (but then that could be finicky, like what if the Jinja isn't _exactly_ the same but is effectively the same).
    - IMO this would be the main reason to go to the effort of implementing auto-generated task IDs for these, but I don't think it'd be a common edge case and it can be manually worked around (e.g. one ETL uses `depends_on_tables_existing` and another ETL uses `depends_on` to also wait for the first ETL's `depends_on_tables_existing` task, or they can just pick unique task IDs as the sensors are so low-impact having duplicates shouldn't really matter).
  - Include the ETL table ID in the generated task ID (but then we'd get super long task IDs that are harder to read and could easily hit Airflow's 250 character limit for task IDs).

So ultimately I decided it was going to be too complex/fragile to try to implement auto-generated task IDs for this.

However, if you have ideas for implementing this functionality in a more straightforward way than I've made it out to be then I'd be happy to hear them!